src/SFA.DAS.AssessorService.Application/Handlers/UserManagement/InviteContactToOrganisationHandler.cs

```diff
@@ -0,0 +1,87 @@
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.AssessorService.Api.Types.Models;
+using SFA.DAS.AssessorService.Api.Types.Models.UserManagement;
+using SFA.DAS.AssessorService.Application.Interfaces;
+using SFA.DAS.AssessorService.Domain.Consts;
+using SFA.DAS.AssessorService.Domain.Entities;
+
+namespace SFA.DAS.AssessorService.Application.Handlers.UserManagement
+{
+    public class InviteContactToOrganisationHandler : IRequestHandler<InviteContactToOrganisationRequest, InviteContactToOrganisationResponse>
+    {
+        private readonly IContactQueryRepository _contactQueryRepository;
+        private readonly IContactRepository _contactRepository;
+        private readonly IContactApplyClient _contactApplyClient;
+        private readonly IMediator _mediator;
+        private readonly ISignInService _signInService;
+        private readonly IOrganisationQueryRepository _organisationQueryRepository;
+
+        public InviteContactToOrganisationHandler(IContactQueryRepository contactQueryRepository,
+            IContactRepository contactRepository,
+            IContactApplyClient contactApplyClient,
+            IMediator mediator, ISignInService signInService, IOrganisationQueryRepository organisationQueryRepository)
+        {
+            _contactQueryRepository = contactQueryRepository;
+            _contactRepository = contactRepository;
+            _contactApplyClient = contactApplyClient;
+            _mediator = mediator;
+            _signInService = signInService;
+            _organisationQueryRepository = organisationQueryRepository;
+        }
+
+        public async Task<InviteContactToOrganisationResponse> Handle(InviteContactToOrganisationRequest request, CancellationToken cancellationToken)
+        {
+            var existingContact = await _contactQueryRepository.GetContactFromEmailAddress(request.Email);
+            if (existingContact != null)
+            {
+                return new InviteContactToOrganisationResponse()
+                {
+                    Success = false,
+                    ErrorMessage = existingContact.OrganisationId == request.OrganisationId
+                        ? "This email address is already registered against your organisation. You must use a unique email address."
+                        : "This email address is already registered against another organisation. You must use a unique email address."
+                };
+            }
+
+            var newContact = await CreateNewContact(request);
```

Comment:
This could be more defensive - there is a lot that could fail here giving nulls and then null references afterwards